### PR TITLE
[bug fix] fixed system notifications not loading

### PR DESCRIPTION
**Pull request by Malyuk-A**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.

### Motivation, Context, and Description
I initially wanted to work on a different issue but noticed/remembered that system notifications are no longer loaded. This issue occurred only recently _(I tested system notifications during the big update PR testing sessions and everything worked fine)_. 

image

So I tried to look into this issue and created a small fix/hack so the system notifications are loaded again.

The error message states that an internal server error occurs in the SystemNotificationResource class. The problem here is that a SQL query is created where one group by value is undefined. This value is taken from the input parameter "pageable" that the client creates. (see image where sort[0] is undefined)
image

Issue is in the client :
The underlying issue is that the router should load specific data (like page) but it is empty/undefined. I do not know the reason for this or how to fix it. It would be great if an expert in this field could look into this.

But the system notifications work as intended again and I only added 2 small if cases to check if the data is undefined or not.
I am not sure if this is the best way to solve this issue so I am curious about your opinions.

### Steps for Testing
Log into Artemis as an Admin (e.g. user 12).
Go to the admin tools -> system notification management page
The system notifications should work just as before.
There should also not be any issues in the developer debugger console.
Make sure related functions still work (e.g. delete, change, display, etc.)

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage
I only added two small if cases that check if the data is undefined or not so no changes to the test coverage.
image

### CODE REVIEW

File: src/main/webapp/app/admin/system-notification-management/system-notification-management.component.ts [hunk 1]
```diff
@@ -135,7 +138,10 @@ export class SystemNotificationManagementComponent implements OnInit, OnDestroy
...
      */
     sort() {
-        const result = [this.predicate + ',' + (this.reverse ? 'asc' : 'desc')];
+        const result = [];
+        if (this.predicate) {
+            result.push(this.predicate + ',' + (this.reverse ? 'asc' : 'desc'));
+        }
         if (this.predicate !== 'id') {
             result.push('id');
         }
```
Comment (Olimbia23) [inline src/main/webapp/app/admin/system-notification-management/system-notification-management.component.ts:147]: [nit] Is this check related to the check above? Should we push `id` even if `this.predicate` is undefined?

**Review by Olimbia23 — APPROVED** [review]
Nice work @Malyuk-A.
Code changes look fine. Left one minor comment.
Approve 👍

File: src/main/webapp/app/admin/system-notification-management/system-notification-management.component.ts (hunk 1 above)
Comment (Malyuk-A) [reply src/main/webapp/app/admin/system-notification-management/system-notification-management.component.ts:147]: This is a good question I am not sure. But because Id should be set if  "predicate" is not equal to "id" it is also the case now because predicate = undefined -> not equal to id. If I understood it correctly it did correctly filter by id on the server so I think this is fine.
